homeassistant/components/bosch_shc/config_flow.py

```diff
@@ -0,0 +1,220 @@
+                    self.hostname = result["token"].split(":", 1)[1]
+
+                await self.hass.async_add_executor_job(
+                    write_tls_asset, self.hass, CONF_SHC_CERT, result["cert"]
+                )
+                await self.hass.async_add_executor_job(
+                    write_tls_asset, self.hass, CONF_SHC_KEY, result["key"]
+                )
+                await validate_input(
+                    self.hass,
+                    self.host,
+                    self.hass.config.path(DOMAIN, CONF_SHC_CERT),
+                    self.hass.config.path(DOMAIN, CONF_SHC_KEY),
+                )
+            except SHCAuthenticationError:
+                errors["base"] = "invalid_auth"
+            except SHCConnectionError:
+                errors["base"] = "cannot_connect"
+            except SHCmDNSError:
+                _LOGGER.warning("Error looking up mDNS entry")
+                errors["base"] = "cannot_connect"
+            except SHCSessionError:
+                _LOGGER.warning("API call returned non-OK result. Wrong password?")
+                errors["base"] = "unknown"
+            except SHCRegistrationError:
+                _LOGGER.warning(
+                    "SHC not in pairing mode! Please press the Bosch Smart Home Controller button until LED starts blinking"
+                )
+                errors["base"] = "unknown"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                return self.async_create_entry(
+                    title=self.info["title"],
+                    data={
+                        CONF_SSL_CERTIFICATE: self.hass.config.path(
+                            DOMAIN, CONF_SHC_CERT
+                        ),
+                        CONF_SSL_KEY: self.hass.config.path(DOMAIN, CONF_SHC_KEY),
+                        CONF_HOST: self.host,
+                        CONF_TOKEN: result["token"],
+                        CONF_HOSTNAME: self.hostname,
+                    },
+                )
+        else:
+            user_input = {}
+
+        schema = vol.Schema(
+            {
+                vol.Required(CONF_PASSWORD, default=user_input.get(CONF_PASSWORD)): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="credentials", data_schema=schema, errors=errors
+        )
+
+    async def async_step_zeroconf(self, zeroconf_info):
+        """Handle zeroconf discovery."""
+        if not zeroconf_info.get("name", "").startswith("Bosch SHC"):
+            return self.async_abort(reason="not_bosch_shc")
+
+        try:
+            self.info = info = await self._get_info(zeroconf_info["host"])
+        except SHCConnectionError:
+            return self.async_abort(reason="cannot_connect")
+        except SHCmDNSError:
+            _LOGGER.exception("Error looking up mDNS entry")
```

Comment:
This could become spammy for the user and the user would not have a good way to avoid it. I'd recommend logging at debug level instead and include the exception info in the log.